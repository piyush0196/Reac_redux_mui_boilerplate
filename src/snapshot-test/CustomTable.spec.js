import TestRenderer from 'react-test-renderer'
import CustomTable from 'components/ui-kit/Table'
import { ThemeProvider } from '@mui/material'
import { theme } from 'theme/theme'

describe('Component - CustomTable', () => {
  const tableHeader = [
    {
      id: 1,
      label: 'Product',
      value: 'product',
      type: 'product',
      isMobFullWidRow: true
    },
    {
      id: 2,
      label: 'Purchased On',
      value: 'purchased_on',
      type: 'date',
      isMobileColumn: true
    },
    {
      id: 3,
      label: 'Buyer',
      value: 'buyer',
      type: 'buyer',
      isMobileColumn: true
    },
    {
      id: 4,
      label: 'Price',
      value: 'price',
      type: 'price',
      isMobileColumn: true
    },
    {
      id: 5,
      label: 'Due By',
      value: 'due_by',
      type: 'date'
    },
    {
      id: 6,
      label: 'Payment date',
      value: 'payment_date',
      type: 'paymentDate'
    },
    {
      id: 8,
      label: 'Status',
      value: 'status',
      type: 'status'
    },
    {
      id: 9,
      label: 'Switch',
      value: 'switch',
      type: 'switch'
    }
  ]

  const tableRows = [
    {
      id: 1,
      product: 'product',
      purchased_on: 'Sept 15, 12:56 PM',
      buyer: 'Ada Reynolds',
      price: '$89',
      due_by: '11-11-2020',
      incomingMessage: 1,
      payment_date: '11-11-2020',
      reviews: 4,
      status: true,
      switch: {
        switchValue: false
      }
    },
    {
      id: 2,
      product: '7 Figure Funnel Strategy by a Sales',
      purchased_on: 'Sept 15, 12:56 PM',
      buyer: 'Ada Reynolds',
      price: '$89',
      due_by: '11-11-2020',
      payment_date: '11-11-2020',
      reviews: 4,
      status: false,
      switch: {
        switchValue: true
      }
    },
    {
      id: 3,
      product: '7 Figure Funnel Strategy by a Sales',
      purchased_on: 'Sept 15, 12:56 PM',
      buyer: 'Ada Reynolds',
      price: '$89',
      due_by: '11-11-2020',
      payment_date: '11-11-2020',
      reviews: 4,
      status: true,
      switch: {
        switchValue: false
      }
    },
    {
      id: 4,
      product: '7 Figure Funnel Strategy by a Sales',
      purchased_on: 'Sept 15, 12:56 PM',
      buyer: 'Ada Reynolds',
      price: '$89',
      due_by: '11-11-2020',
      payment_date: '11-11-2020',
      reviews: 4,
      status: false,
      switch: {
        switchValue: true
      }
    }
  ]

  const TableJSX = ({
    type,
    rows,
    header,
    currentPage,
    totalPages,
    onPageChange
  }) => {
    return (
      <ThemeProvider theme={theme}>
        <CustomTable
          rows={rows}
          header={header}
          currentPage={currentPage}
          totalPage={totalPages}
          onPageChange={onPageChange}
          type={type}
        />
      </ThemeProvider>
    )
  }

  const wrapper1 = TestRenderer.create(<TableJSX header={tableHeader} row={tableRows} />)
  const wrapper2 = TestRenderer.create(
    <TableJSX header={tableHeader} row={tableRows} type='static' />
  )
  const wrapper3 = TestRenderer.create(
    <TableJSX
      header={tableHeader}
      row={tableRows}
      currentPage={1}
      totalPage={2}
      onPageChange={() => {}}
    />
  )
  const wrapper4 = TestRenderer.create(<TableJSX header={tableHeader} row={tableRows} type='static' />)

  it('renders Table with default variant correctly', () => {
    expect(wrapper1).toMatchSnapshot()
  })

  it('renders Table with static variant correctly', () => {
    expect(wrapper2).toMatchSnapshot()
  })

  it('renders Table with pagination correctly', () => {
    expect(wrapper3).toMatchSnapshot()
  })

  it('renders Table without pagination correctly', () => {
    expect(wrapper4).toMatchSnapshot()
  })
})
