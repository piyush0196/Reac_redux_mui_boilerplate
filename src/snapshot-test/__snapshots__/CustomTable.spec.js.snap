// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - CustomTable renders Table with default variant correctly 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-1t8xe81-MuiPaper-root-MuiTableContainer-root"
    size="small"
    type="default"
  >
    <table
      aria-label="simple table"
      className="MuiTable-root css-8fcqlw-MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root css-1vn4xlj-MuiTableHead-root"
        role={null}
        type="default"
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head css-2hu3kr-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1lp87cm-MuiTableCell-root"
            scope="col"
            type="default"
          >
            Product
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1lp87cm-MuiTableCell-root"
            scope="col"
            type="default"
          >
            Purchased On
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1lp87cm-MuiTableCell-root"
            scope="col"
            type="default"
          >
            Buyer
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1lp87cm-MuiTableCell-root"
            scope="col"
            type="default"
          >
            Price
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1lp87cm-MuiTableCell-root"
            scope="col"
            type="default"
          >
            Due By
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1lp87cm-MuiTableCell-root"
            scope="col"
            type="default"
          >
            Payment date
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1lp87cm-MuiTableCell-root"
            scope="col"
            type="default"
          >
            Status
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1lp87cm-MuiTableCell-root"
            scope="col"
            type="default"
          >
            Switch
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        role={null}
      />
    </table>
    <div
      className="css-padqdi"
    >
      No data avalaible
    </div>
    <div
      className="css-mi4t00"
      data={0}
    />
  </div>
</div>
`;

exports[`Component - CustomTable renders Table with pagination correctly 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-1t8xe81-MuiPaper-root-MuiTableContainer-root"
    size="small"
    type="default"
  >
    <table
      aria-label="simple table"
      className="MuiTable-root css-8fcqlw-MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root css-1vn4xlj-MuiTableHead-root"
        role={null}
        type="default"
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head css-2hu3kr-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1lp87cm-MuiTableCell-root"
            scope="col"
            type="default"
          >
            Product
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1lp87cm-MuiTableCell-root"
            scope="col"
            type="default"
          >
            Purchased On
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1lp87cm-MuiTableCell-root"
            scope="col"
            type="default"
          >
            Buyer
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1lp87cm-MuiTableCell-root"
            scope="col"
            type="default"
          >
            Price
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1lp87cm-MuiTableCell-root"
            scope="col"
            type="default"
          >
            Due By
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1lp87cm-MuiTableCell-root"
            scope="col"
            type="default"
          >
            Payment date
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1lp87cm-MuiTableCell-root"
            scope="col"
            type="default"
          >
            Status
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1lp87cm-MuiTableCell-root"
            scope="col"
            type="default"
          >
            Switch
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        role={null}
      />
    </table>
    <div
      className="css-padqdi"
    >
      No data avalaible
    </div>
    <div
      className="css-mi4t00"
      data={0}
    />
  </div>
</div>
`;

exports[`Component - CustomTable renders Table with static variant correctly 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-xu08jz-MuiPaper-root-MuiTableContainer-root"
    size="small"
    type="static"
  >
    <table
      aria-label="simple table"
      className="MuiTable-root css-8fcqlw-MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root css-14pv9g6-MuiTableHead-root"
        role={null}
        type="static"
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head css-2hu3kr-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-13hy3ze-MuiTableCell-root"
            scope="col"
            type="static"
          >
            Product
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-13hy3ze-MuiTableCell-root"
            scope="col"
            type="static"
          >
            Purchased On
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-13hy3ze-MuiTableCell-root"
            scope="col"
            type="static"
          >
            Buyer
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-13hy3ze-MuiTableCell-root"
            scope="col"
            type="static"
          >
            Price
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-13hy3ze-MuiTableCell-root"
            scope="col"
            type="static"
          >
            Due By
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-13hy3ze-MuiTableCell-root"
            scope="col"
            type="static"
          >
            Payment date
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-13hy3ze-MuiTableCell-root"
            scope="col"
            type="static"
          >
            Status
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-13hy3ze-MuiTableCell-root"
            scope="col"
            type="static"
          >
            Switch
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        role={null}
      />
    </table>
    <div
      className="css-padqdi"
    >
      No data avalaible
    </div>
    <div
      className="css-mi4t00"
      data={0}
    />
  </div>
</div>
`;

exports[`Component - CustomTable renders Table without pagination correctly 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-xu08jz-MuiPaper-root-MuiTableContainer-root"
    size="small"
    type="static"
  >
    <table
      aria-label="simple table"
      className="MuiTable-root css-8fcqlw-MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root css-14pv9g6-MuiTableHead-root"
        role={null}
        type="static"
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head css-2hu3kr-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-13hy3ze-MuiTableCell-root"
            scope="col"
            type="static"
          >
            Product
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-13hy3ze-MuiTableCell-root"
            scope="col"
            type="static"
          >
            Purchased On
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-13hy3ze-MuiTableCell-root"
            scope="col"
            type="static"
          >
            Buyer
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-13hy3ze-MuiTableCell-root"
            scope="col"
            type="static"
          >
            Price
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-13hy3ze-MuiTableCell-root"
            scope="col"
            type="static"
          >
            Due By
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-13hy3ze-MuiTableCell-root"
            scope="col"
            type="static"
          >
            Payment date
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-13hy3ze-MuiTableCell-root"
            scope="col"
            type="static"
          >
            Status
          </th>
          <th
            aria-sort={null}
            cellin="header"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-13hy3ze-MuiTableCell-root"
            scope="col"
            type="static"
          >
            Switch
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        role={null}
      />
    </table>
    <div
      className="css-padqdi"
    >
      No data avalaible
    </div>
    <div
      className="css-mi4t00"
      data={0}
    />
  </div>
</div>
`;
